src/test/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjectionTest.java

```diff
@@ -0,0 +1,114 @@
+
+        // Real localhost ping is not able to operate in CI, Mock instead
+        doReturn(stringBuilder)
+                .when(commandInjection)
+                .getResponseFromPingCommand(eq(localhost), any(Supplier.class));
+
+        doReturn(trueCondition).when(commandInjection).getSupplier(true);
+
+        doReturn(falseCondition).when(commandInjection).getSupplier(false);
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_Localhost_ExpectPingSuccess() throws IOException {
+        // Act
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(localhost);
+
+        // Assert
+        Assert.isTrue(
+                successReturnStr.equals(responseEntity.getBody().getContent()),
+                "must get response");
+        Assert.isTrue(HttpStatus.OK == responseEntity.getStatusCode(), "must equal");
+        Assert.isTrue(responseEntity.getBody().getIsValid(), "must equal");
+
+        verify(commandInjection).getVulnerablePayloadLevel1(localhost);
+        verify(commandInjection).getResponseFromPingCommand(eq(localhost), eq(trueCondition));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_NoIp_ExpectNoResponseContent() throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(Strings.EMPTY);
+
+        // Assert
+        Assert.isTrue(StringUtils.isBlank(responseEntity.getBody().getContent()), "must be empty");
+        Assert.isTrue(HttpStatus.OK == responseEntity.getStatusCode(), "must equal");
+        Assert.isTrue(responseEntity.getBody().getIsValid(), "must equal");
+
+        verify(commandInjection).getVulnerablePayloadLevel1(eq(Strings.EMPTY));
+        verify(commandInjection).getResponseFromPingCommand(eq(Strings.EMPTY), eq(falseCondition));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_Localhost_ExpectPingSuccess()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity = new RequestEntity(HttpMethod.GET, URI.create(localhost));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(localhost, entity);
+
+        Assert.isTrue(
+                successReturnStr.equals(responseEntity.getBody().getContent()),
+                "must get response");
+        Assert.isTrue(HttpStatus.OK == responseEntity.getStatusCode(), "must equal");
+        Assert.isTrue(responseEntity.getBody().getIsValid(), "must equal");
+
+        verify(commandInjection).getVulnerablePayloadLevel2(eq(localhost), eq(entity));
+        verify(commandInjection).getResponseFromPingCommand(eq(localhost), eq(trueCondition));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_WrongHost_Return_NotResolve()
```

Comment:
I think these test cases are quite similar and as business logic we should test is:
1. ``` Supplier<Boolean> condition =
                () ->
                        ipAddress != null
                                && !SEMICOLON_SPACE_LOGICAL_AND_PATTERN
                                        .matcher(requestEntity.getUrl().toString())
                                        .find();
 i.e. predicate condition
2. Same response and "\n".

So i think you can remove this method. and for each level you can add test cases for Predicate conditions.